Sqlite: Fix deletion for larger numbers of checkpoints/transactions

rvl opened the pull request:
Relates to #154.
Base branch is #341 , so that needs to be merged first.

# Overview

Any instance of `deleteWhere` and `<-.` together is potentially a place where there will be too many variables in the SQL statement.

This removes those instances and replaces them with batched deletes (`deleteMany`), or where that's not possible, a delete from a join.

## Comments

- [ ] TODO: regression test

denisshevchenko commented on src/Cardano/Wallet/Kernel/Read.hs:
```diff
@@ -60,96 +54,69 @@ getWalletSnapshot pw = query' (pw ^. wallets) Snapshot
+        :: DB
+        -> HdAccountId
+        -> [(Address, Word32)]
+    addressesByAccountId' _db accId = addressesByAccountId db accId
```
If the first argument is not used - why do we need it? Maybe just remove `_db`?

denisshevchenko commented on src/Cardano/Wallet/Kernel/Read.hs:
```diff
@@ -60,96 +54,69 @@ getWalletSnapshot pw = query' (pw ^. wallets) Snapshot
+    -- NOTE:
+    -- We arbitrarily take the maximum AddressPoolGap here. This is rather
+    -- incorrect but will go away with the next PR as part of #357!
+    externallyOwnedRoots
```
Maybe we have to unify our naming? I mean that in some places we use a full form `externallyOwned`, but in other places we use a short one, `EO`...

denisshevchenko commented on src/Cardano/Wallet/Kernel/BListener.hs:
```diff
@@ -78,9 +79,20 @@ prefilterContext
+    invariantFO :: HdRootId -> Maybe a -> IO a
+    invariantFO rootId = flip maybe return $ fail $ toString $
+        "Cardano.Wallet.Kernel.BListener: invariant violation: encrypted secret key \
+        \hasn't been found for the given root id: " <> sformat build rootId
```
Maybe just match explicitly?

```
    invariantFO :: HdRootId -> Maybe a -> IO a
    invariantFO rootId (Just k) = return k
    invariantFO rootId Nothing = fail $ toString $
        "Cardano.Wallet.Kernel.BListener: invariant violation: encrypted secret key \
        \hasn't been found for the given root id: " <> sformat build rootId
```

rvl committed:
Sqlite: use deleteMany where deleting checkpoints

rvl committed:
Sqlite: Use a join for deleteLooseTransactions

It would have exploded if there were more than 999 TxMeta or
PendingTx.

I tried Esqueleto and it can't do this query. Actually, it could
express the query, but the generated SQL was invalid. And the code
looked just as bad as raw SQL.
